Source/Render/sokol/SokolTypes.h

```diff
@@ -1,34 +1,38 @@
 #ifndef PERIMETER_SOKOLTYPES_H
 #define PERIMETER_SOKOLTYPES_H
 
+#include <tuple>
+
 #include "RenderTypes.h"
 
 enum SOKOL_SHADER_ID {
     SOKOL_SHADER_ID_NONE,
+    SOKOL_SHADER_ID_tex1,
     SOKOL_SHADER_ID_color_tex1,
     SOKOL_SHADER_ID_color_tex2,
     SOKOL_SHADER_ID_normal,
-    SOKOL_SHADER_ID_terrain,
+    SOKOL_SHADER_ID_object_shadow,
+    SOKOL_SHADER_ID_only_texture,
+    SOKOL_SHADER_ID_tile_map,
 };
 
 enum PIPELINE_TYPE {
-    PIPELINE_TYPE_TRIANGLE,
-    PIPELINE_TYPE_TRIANGLESTRIP,
-    PIPELINE_TYPE_TERRAIN,
+    PIPELINE_TYPE_MESH,
+    PIPELINE_TYPE_TILE_MAP,
+    PIPELINE_TYPE_OBJECT_SHADOW,
 #ifdef PERIMETER_DEBUG
     PIPELINE_TYPE_LINE_STRIP,
 #endif
     PIPELINE_TYPE_MAX,
 };
-const PIPELINE_TYPE PIPELINE_TYPE_DEFAULT = PIPELINE_TYPE_TRIANGLE;
+const PIPELINE_TYPE PIPELINE_TYPE_DEFAULT = PIPELINE_TYPE_MESH;
 
 
 enum SOKOL_MATERIAL_TYPE {
     SOKOL_MAT_NONE = 0,
     SOKOL_MAT_LIGHT = 1,
 };
 
-using pipeline_id_t = uint32_t;
```

Comment:
Due to complexity of adding new pipeline types (and bit shifting) I decided to search pipeline by combination of parameters from context. I keep all pipelines in vector and since we don't expect to have a lot of pipelines, search should not be a problem.